runtime/lua/vim/iter.lua

```diff
@@ -179,6 +208,60 @@ function ListIter.filter(self, f)
   return self
 end
 
+--- Flattens a |list-iterator|, un-nesting nested values up to the given {depth}.
+--- Errors if it attempts to flatten a dict-like value.
+---
+--- Example:
+---
+--- ```lua
+--- vim.iter({1, {2}, {{3}}}):flatten():totable()
+--- ```
+--- (returns `{1, 2, {3}}`)
+---
+--- ```lua
+--- vim.iter({1, {{a = 2}}, {3}}):flatten():totable()
+--- ```
+--- (returns `{1, {a = 2}, 3}`)
```

Comment:
```suggestion
--- ```lua
--- vim.iter({1, {{a = 2}}, {3}}):flatten():totable()
--- -- {1, {a = 2}, 3}
--- ```
```